compiler/src/Language/Bond/Codegen/Cpp/Reflection_h.hs

```diff
@@ -106,16 +106,25 @@ reflection_h export_attribute cpp file imports declarations = ("_reflection.h",
         fieldMetadata Field {..} =
             [lt|private: #{export_attr}static const ::bond::Metadata s_#{fieldName}_metadata;|]
 
-        fieldTemplates = F.foldMap $ \ f@Field {..} -> [lt|
+        -- fieldTemplateReservedNames are static variables names used in ::bond::reflection::FieldTemplate<>
+        fieldTemplateReservedNames = ["metadata", "field", "id"]
+        
+        fieldNames = map (\f -> fieldName f) structFields
+
+        fieldTemplateStructReservedNames = fieldTemplateReservedNames ++ fieldNames
+
+        uniqueFieldTemplateStructNames = uniqueNames (map (\n -> "_" ++ n) fieldNames) fieldTemplateStructReservedNames
```

Comment:
We should also avoid using [reserved identifiers](https://en.cppreference.com/w/cpp/language/identifiers).
> the identifiers with a double underscore anywhere are reserved;
> the identifiers that begin with an underscore followed by an uppercase letter are reserved;
> the identifiers that begin with an underscore are reserved in the global namespace.

Reply:
Addressed in latest revision.